db/migrate/20210406060434_fix_topic_user_bookmarked_sync_issues_again.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class FixTopicUserBookmarkedSyncIssuesAgain < ActiveRecord::Migration[6.0]
+  def up
+    DB.exec(
+      <<~SQL
+        UPDATE topic_users SET bookmarked = true
+        FROM bookmarks AS b
+        WHERE NOT topic_users.bookmarked AND topic_users.topic_id = b.topic_id AND topic_users.user_id = b.user_id
+      SQL
+    )
+
+    DB.exec(
+      <<~SQL
+        UPDATE topic_users SET bookmarked = false
+        WHERE topic_users.bookmarked AND (SELECT COUNT(*) FROM bookmarks WHERE topic_id = topic_users.topic_id AND user_id = topic_users.user_id) = 0
```

Comment:
Overall the PR looks good, but I'm always unsure of performance implications of queries like this. 😃 

It will run a `count(*)` subquery for each `topic_users` record that has `bookmarked` set to `true`? Will the database optimize it, and even if it will – are you sure it won't cause problems on massive forums?

The same question applies to `SyncTopicUserBookmarked`.

Reply:
Well, I assume it will work fine because a similar migration has been run twice before without any issues :) See the other 2 PRs linked in the OP. I will do a quick test against meta and see (using  BEGIN...ROLLBACK)

Reply:
@CvX I just tried this on meta, which has 5,423,106 topic_user records. The first query took ~1s, the second query took ~4s.